renaissance-core/src/main/java/org/renaissance/core/BenchmarkInfo.java

```diff
@@ -0,0 +1,104 @@
+package org.renaissance.core;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public final class BenchmarkInfo {
+
+    final String className;
```

Comment:
Wrong indentation. I mean, 4 spaces is the right indentation but we use 2 spaces everywhere else 😉.

Reply:
Ooops! :-)

Reply:
> > * Support for named configurations, inspired by @vhotspur and #182
> 
> And where is my `dummy-params` benchmark? cry

Nothing personal, forgot about it once I converted all the benchmarks. I guess I can still add it :-)

> I would eventually like to see parameters as annotations on fields where the value would be injected. I think `intParameter` and `doubleParameter` are fine.

I think we are on the same page here. To make it easy to see what different configurations override, I would keep the `@Configuration` annotation per class (to list overrides), and use the `@Parameter` annotation per field, picking name and type from the field and allowing to set default value and description. However, not sure how we would handle list parameter types.

> 
> > * Let me know if you think we should include a method like `stringListParameter` or `intListParameter` or `doubleListParameter`.
> 
> It is already on 4 places so probably yes just to ensure that we use comma consistently everywhere.

Yeah, forgetting to trim the split values has bitten me a few times. Having a common method could have prevented that.


> > * This is just an initial sketch, tested so far only on the `ResultWriter`, which is registered to receive some of the events, allowing to remove the old `ResultObserver` interface. As we try to develop plugins, we will see whether this is sufficient.
> 
> Unless the plugin loading (if given on command-line) code is already in the pipeline, I can try to add that.

You are welcome to give it a shot.

> 
> > Not many, but I would like to change the method names int the `Benchmark` interface, so that we make them a bit more streamlined and we get rid of the word "iteration". 
> Personally, I would go with JUnit naming as that is quite familiar to Java user and use `beforeAll`, `before`, `run`, `after` and `afterAll`.

Or that. I was trying to be a bit more verbose, because the `Benchmark` is more like a wrapper for a measured operation, and when you call `benchmark.beforeAll()`, it also "runs", but not the measured operation. Only the `run` method has a verb indicating an action (in this line of reasoning, my previous `operation()` should really be `runOperation`. So I would argue to include verbs in the methods, e.g., at least setUpBeforeAll, setUp, run, tearDown, tearDownAfterAll, which is not so different from the current state, but gets rid of the word "iteration. I apologize for my OCD, I guess we are splitting hairs here :-)

> > An alternative is to consider returning it from `afterIteration/tearDownOperation`, which is outside the measured block. This would however require the benchmark to keep the result around until the method is called and actually require a benchmark to implement at least two methods instead of one.
> 
> I am against splitting this. Instantiation of a `BenchmarkResult` class should have negligible effect as the user would typically store the compute result in one field and return that class (and if that creates an unwanted overhead, it is not a benchmark we want, I guess). Any computations can be (and should be) done in the `validate()` method.

Fine with me.

> 
> > * Plugins/event listeners currently cannot publish custom results. We will need to provide some kind of callback interface that these listeners can use to publish their results (e.g., hardware counters).
> 
> For me this is the biggest issue as it prevents us to create plugin for [ubench-agent](https://github.com/D-iii-S/java-ubench-agent).

Yes, I know. But I think it should be easier to add now. If you can try to play with loading of plugins and registering them with the right listener list (in the right order), we can try to provide an interface for publishing results in one of the events.

> 
> > Please give it a try and be gentle :-)
> 
> I all for merging this in the current state.

Thanks :-)